raw/mock_exam/docs/questions/question-32.adoc

```diff
@@ -0,0 +1,105 @@
+// tag::DE[]
+=== Frage 32
+**ID: Q-20-04-25**
+
+[cols="2,8,2", frame=ends, grid=rows]
+|===
+|K-Frage: 
+|Wählen Sie für jede Zeile „richtig“ oder „falsch“ aus.
+| 2 Punkte
+|===
+
+Was sind Richtlinien für ein gutes Schnittstellen-Design?
+Kreuzen Sie an, welche der folgenden Aussagen richtig und welche falsch sind.
+
+
+[cols="2a,2a,1, 7", frame=none, grid=none]
+|===
+
+| richtig
+| falsch
+|
+|
+
+| {y} 
+| {n}
+| (a)
+| Die Verwendung der Schnittstellen sollte einfach zu erlernen sein.
+
+| {y}
+| {n}
+| (b)
+| Der Client-Code sollte angemessen leicht zu verstehen sein.
+
+| {n}
+| {y}
+| (c)
+| Eine Schnittstelle soll Zugriff auf eine umfassende Menge von Implementierungsdetails ermöglichen
+
+| {y}
+| {n}
+| (d)
+| Schnittstellenspezifikationen sollten funktionale und nichtfunktionale Aspekte enthalten.
+
+| {n}
+| {y}
+| (e)
+| Eine Schnittstelle soll die Implementierung so abstrahieren, dass eine Unterscheidung zwischen lokalen und entfernten Aufrufen nicht nötig ist.
```

Comment:
Verstehe Sinn dieser Antwortmöglichkeit nicht wirklich.

Reply:
Ich kann den Hintergrund erklären, aber ich glaube es wäre schneller einen Verbesserung für eine neue Formulierung zu geben.  Wie wäre bspw.:

"Schnittstellen, die (ggf. geographisch) verteilten Server aufrufen sollen genau so implementiert werden, wie Schnittstellen, die lokalen Server aufrufen."

Die Idee dahinter ist, dass Qualitätseigenschaften wie Performance, Latzenzeit etc. müssen mehr für remote Aufrufe berücksichtigt werden als für lokale aufrufen.

Reply:
Ja, also bei der Intention bin ich bei Dir und finde die gut. Deine neue Formulierung finde ich verständlicher.

Reply:
nope, I'm not convinced of the updated proposal. I personally don't understand it (sorry)

The background of this option is to distinguish between local and remote. 

Changed to 
Lokale und entfernte ("remote") Aufrufe dieser Schnittstelle sollen sich in allen Belangen identisch verhalten.